lib/Echidna/Transaction.hs

```diff
@@ -63,38 +63,38 @@ makeLenses 'TxConf
 ppSolCall :: SolCall -> String
 ppSolCall (t, vs) = (if t == "" then T.unpack "*fallback*" else T.unpack t) ++ "(" ++ intercalate "," (ppAbiValue <$> vs) ++ ")"
 
+instance ToJSON Tx where
+  toJSON (Tx c s d g v (t, b)) = object [ ("call",        toJSON $ either ppSolCall (const "<CREATE>") c)
+                                        -- from/to are Strings, since JSON doesn't support hexadecimal notation
+                                        , ("from",        toJSON $ show s)
+                                        , ("to",          toJSON $ show d)
+                                        , ("value",       toJSON $ show v)
+                                        , ("gas",         toJSON $ show g)
+                                        , ("time delay",  toJSON $ show t)
+                                        , ("block delay", toJSON $ show b)
+                                        ]
+
 -- | If half a tuple is zero, make both halves zero. Useful for generating delays, since block number
 -- only goes up with timestamp
 level :: (Num a, Eq a) => (a, a) -> (a, a)
 level (elemOf each 0 -> True) = (0,0)
 level x                       = x
 
-instance ToJSON Tx where
-  toJSON (Tx c s d g v (t,b)) = object [ ("call",        toJSON $ either ppSolCall (const "<CREATE>") c)
-                                       -- from/to are Strings, since JSON doesn't support hexadecimal notation
-                                       , ("from",        toJSON $ show s)
-                                       , ("to",          toJSON $ show d)
-                                       , ("value",       toJSON $ show v)
-                                       , ("gas",         toJSON $ show g)
-                                       , ("time delay",  toJSON $ show t)
-                                       , ("block delay", toJSON $ show b)
-                                       ]
-
 -- | A contract is just an address with an ABI (for our purposes).
 type ContractA = (Addr, [SolSignature])
 
 -- | The world is made our of humans with an address, and contracts with an address + ABI.
 data World = World { _senders   :: [Addr]
                    , _receivers :: [ContractA]
                    }
-
+  deriving (Show)
```

Comment:
are we using this instance? if so, we should probably make it somewhat more clear by writing it explicitly

Reply:
we don't use this instance but it probably doesn't hurt to have right?

Reply:
if we're going to ship an instance, it should be a good instance. I don't think the auto-derived one is good